applications: asset_tracker_v2: Add documentation for the cloud module (#6315)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst

Overview
********

> The application samples sensor data and publishes the data to a connected cloud service over `TCP/IP`_ through `LTE`_.

Author: I was referring to the more generic term TCP/IP (Internet Protocol Suite) that also includes UDP.

> The application samples sensor data and publishes the data to a connected cloud service over `TCP/IP`_ through `LTE`_.

Reviewer: Well, TCP does not "include" UDP.

> The application samples sensor data and publishes the data to a connected cloud service over `TCP/IP`_ through `LTE`_.

Author: "TCP/IP" does. It's a common name used to reference protocols used on the web. If you think it's misleading we can change it to something else.

> The application samples sensor data and publishes the data to a connected cloud service over `TCP/IP`_ through `LTE`_.

Reviewer: Disagree.

```suggestion
The application samples sensor data and publishes the data to a connected cloud service over `IP`_ through `LTE`_.
```

|          | Movement timeout    | Number of seconds between each sampling/publication in passive mode, regardless if the device is moving or not.                      | 3600 seconds   |
+----------+---------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+
| GPS timeout                    | Timeout for acquiring a GNSS fix during data sampling.                                                                               | 60 seconds     |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+
| Accelerometer threshold        | Accelerometer threshold in m/s². Minimal absolute value in m/s² for accelerometer readings to be considered valid movement.          | 10 m/s²        |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+

> | No Data List (NOD)             | List that contains :ref:`data types <app_data_types>` that is not sampled by the application. Used to disable sampling from          | No entries     |

Author: Datatypes are not sampled

|          | Movement timeout    | Number of seconds between each sampling/publication in passive mode, whether the device is moving or not.                            | 3600 seconds   |
+----------+---------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+
| GPS timeout                    | Timeout for acquiring a GNSS fix during data sampling.                                                                               | 60 seconds     |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+
| Accelerometer threshold        | Accelerometer threshold in m/s². Minimal absolute value in m/s² for accelerometer readings to be considered valid movement.          | 10 m/s²        |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------+

> | No Data List (NOD)             | List that contains :ref:`data types <app_data_types>` that is not sampled by the application. Used to disable sampling from          | No entries     |

Reviewer 2: ```suggestion
| No Data List (NOD)             | A list containing :ref:`data types <app_data_types>` that are not sampled by the application. Used to disable sampling from          | No entries     |
```

These options can be used to enable and disable modules and modify their behavior and properties.
|config|

Setting up the Asset Tracker cloud example
==========================================

To set up the application to work with a specific cloud example, see the following documentation:

* nRF Cloud - `Creating an nRF Cloud account`_ and `Connecting your device to nRF Cloud`_.
* AWS IoT Core - `Getting started guide for nRF Asset Tracker for AWS`_
* Azure IoT Hub - `Getting started guide for nRF Asset Tracker for Azure`_

By default, the application is configured to communicate with `nRF Cloud`_ using the factory-provisioned certificates on Thingy:91 and nRF9160 DK.
This enables the application to function out-of-the-box with nRF Cloud.
However, nRF Cloud does not fully support the application firmware and has limitations.
For more information, see :ref:`nrf_cloud_limitations`.
To enable all features of the Asset Tracker v2, use any of the other supported cloud service implementations.

.. note::

>    Before building and running the firmware, ensure that the cloud side is set up and the device has been provisioned with the correct SSL certificates.

Reviewer: ```suggestion
   Before building and running the firmware, ensure that the cloud side is set up and the device has been provisioned with the correct TLS certificates.
```

CONFIG_ASSET_TRACKER_V2_APP_VERSION - Configuration for providing the application version
   The application publishes its version number as a part of the static device data. The default value for the application version is ``0.0.0-development``. To configure the application version, set :ref:`CONFIG_ASSET_TRACKER_V2_APP_VERSION <CONFIG_ASSET_TRACKER_V2_APP_VERSION>` to the desired version.

.. _default_config_values:

> The default values for the device configuration values can be set compile time by manipulating the following configurations:

Reviewer: ```suggestion
The default values for the device configuration values can be set at compile time by manipulating the following configurations:
```

DTS overlay files are named the same as the build target and use the file extension :file:`.overlay`.
They are placed in the :file:`boards` folder.
When the DTS overlay filename matches the build target, the overlay is automatically chosen and applied by the build system.

> An overlay file typically enable a specific feature and can be included with over overlay files to enable multiple features at the same time.

Reviewer: ```suggestion
An overlay file typically enables a specific feature and can be included with over overlay files to enable multiple features at the same time.
```
